xarray/plot/utils.py

```diff
@@ -73,13 +74,19 @@ def _color_palette(cmap, n_colors):
         # we have a list of colors
         try:
             # first try to turn it into a palette with seaborn
-            from seaborn.apionly import color_palette
-            pal = color_palette(cmap, n_colors=n_colors)
+            with warnings.catch_warnings():
+                warnings.filterwarnings('error')
+                try:
+                    from seaborn.apionly import color_palette
+                except UserWarning:
```

Comment:
Let's make this `except (UserWarning, ImportError)`, to guard against when `seaborn.apionly` is removed in some future version.